src/Model/Post.php

```diff
@@ -0,0 +1,390 @@
+		 * If the post_type isn't (not registered) or is not allowed in WPGraphQL,
+		 * mark the post as private
+		 */
+		if ( empty( $this->post_type_object ) || empty( $this->post_type_object->name ) || ! in_array( $this->post_type_object->name, \WPGraphQL::$allowed_post_types, true ) ) {
+			return true;
+		}
+
+		if ( 'private' === $data->post_status && ! current_user_can( $this->post_type_object->cap->read_private_posts ) ) {
+			return true;
+		}
+
+		if ( 'revision' === $data->post_type ) {
+			$parent               = get_post( (int) $data->post_parent );
+			$parent_post_type_obj = get_post_type_object( $parent->post_type );
+			if ( ! current_user_can( $parent_post_type_obj->cap->edit_post, $parent->ID ) ) {
+				return true;
+			}
+		}
+
+		return $private;
+
+	}
+
+	/**
+	 * Initialize the Post object
+	 *
+	 * @param null|string|array $filter The field or fields to build in the modeled object. You can
+	 *                                  pass null to build all of the fields, a string to only
+	 *                                  build an object with one field, or an array of field keys
+	 *                                  to build an object with those keys and their respective
+	 *                                  values.
+	 *
+	 * @access public
+	 * @return void
+	 */
+	public function init( $filter = null ) {
+
+		if ( 'private' === parent::get_visibility() ) {
+			return null;
+		}
+
+		if ( empty( $this->fields ) ) {
+			$this->fields = [
+				'ID' => function() {
+					return $this->post->ID;
+				},
+				'post_author'   => function() {
+					return ! empty( $this->post->post_author ) ? $this->post->post_author : null;
+				},
+				'id'            => function () {
+					return ( ! empty( $this->post->post_type ) && ! empty( $this->post->ID ) ) ? Relay::toGlobalId( $this->post->post_type, $this->post->ID ) : null;
+				},
+				'post_type'     => function() {
+					return isset( $this->post->post_type ) ? $this->post->post_type : null;
+				},
+				'ancestors'     => function () {
+					$ancestor_ids = get_ancestors( $this->post->ID, $this->post->post_type );
+					return ( ! empty( $ancestor_ids ) ) ? $ancestor_ids : null;
+				},
+				'author'        => function () {
+					$id     = $this->post->post_author;
+					$author = isset( $id ) ? DataSource::resolve_user( absint( $this->post->post_author ) ) : null;
+					return $author;
+				},
+				'date'          => function () {
+					return ! empty( $this->post->post_date ) && '0000-00-00 00:00:00' !== $this->post->post_date ? $this->post->post_date : null;
+				},
+				'dateGmt'       => function () {
+					return ! empty( $this->post->post_date_gmt ) ? Types::prepare_date_response( $this->post->post_date_gmt ) : null;
+				},
+				'contentRendered' => function() {
+					$content = ! empty( $this->post->post_content ) ? $this->post->post_content : null;
+					return ! empty( $content ) ? apply_filters( 'the_content', $content ) : null;
+				},
+				'contentRaw' => function() {
```

Comment:
So, I think there's still some gray area to restricted. 

Like the RAW content should only be accessible to authenticated users. So even if the post is considered Public, the raw content is still restricted if the request isn't authenticated. 

The REST API handles this via an `edit` context. 

I think we can handle it via a simple `if ( empty( get_current_user_id() ) ) { return null; }` check, or something to that tune

I think some fields will need more precise treatment than the 3 common `public`, `restricted` and `private` states of the object at large.

And these cap checks should happen at the Model, not the GraphQL resolvers.